### ISSUE TITLE
Cleanup X2UnifiedProjectile

### ISSUE THREAD

**Issue by Iridar** [X2CommunityCore/X2WOTCCommunityHighlander#1070]
According to our previous tests done for #720, we discovered that instances of `X2UnifiedProjectile` actor remain active forever until the world is destroyed by loading a save or ending a tactical mission, causing a performance impact, and in general there's no reason for them to exist.

As such they should be cleaned up manually, probably by expanding on the solution proposed for #1061 - have an overseer Actor that will return particle systems to the pool.

**Comment by Iridar** [comment]
TODO: go through game's code and list all non-temporary references to x2up just to see what we could break if we Destroy() it.

**Comment by Xymanek** [comment]
Apparently I derped hard when looking through code and X2UPs do destroy themselves:

https://github.com/X2CommunityCore/X2WOTCCommunityHighlander/blob/de17827ec72dd9662170c96b8f74c41d534bf212/X2WOTCCommunityHighlander/Src/XComGame/Classes/X2UnifiedProjectile.uc#L2268

However, the object count in the #720 dumps of X2UPs still has increased:

![image](https://user-images.githubusercontent.com/2865341/132938036-564b40aa-d370-46c2-993b-d4391fa26ec9.png)

![image](https://user-images.githubusercontent.com/2865341/132938038-55366f9e-2112-4161-95de-ed4aeba601e7.png)

It's possible that the existing self-destruct code doesn't cover all cases.

**Comment by Iridar** [comment]
Was #1072 supposed to close this issue, or is it a separate thing? Is this still relevant?

**Comment by Xymanek** [comment]
No, this is a separate thing. However, X2UP might have been not cleaning up itself properly due to PSCs not being cleaned up properly - we should retest with all the latest fixes and see if the problem is still there

**Comment by Xymanek** [comment]
https://youtu.be/uUsoQbylkFA?t=6257 (CHL v1.23.2) still exhibits FPS slowdowns that are solved by save & reload. It might be X2UPs or something else (or both) but the problem definitely still exists (although (I assume) to a lesser degree)
